extensions/src/main/java/org/pentaho/platform/plugin/services/cache/LastModifiedCache.java

```diff
@@ -78,13 +79,11 @@ public <T> T unwrap(Class<T> cls) {
 
   @Override public Set<Object> getAllKeys() {
     Cache<Object, Object> cache = ( (JCacheAccessImpl) getStorageAccess() ).getUnderlyingCache();
-    Set<Object> keys = new HashSet<>();
+    Set<Object> keys = ConcurrentHashMap.newKeySet();
     if ( cache != null ) {
-      for ( Cache.Entry<Object, Object> entry : cache ) {
-           keys.add( entry.getKey() );
-      }
+      cache.forEach(entry -> keys.add( entry.getKey() ) );
```

Comment:
I think this we can replace with return new HashSet<>(cache.keySet()) instead of 2 lines